dali/base/dadfs.cpp

```diff
@@ -1396,15 +1398,27 @@ class CDelayedDelete: public CInterface
                 {
                     case SDSExcpt_LockTimeout:
                     case SDSExcpt_LockHeld:
-                        e->Release();
+                        timeoutException.setown(e);
                         break;
                     default:
                         throw;
                 }
             }
             file.clear();
+            unsigned sleepTime = SDS_TRANSACTION_RETRY/2+(getRandom()%SDS_TRANSACTION_RETRY);
+            if (INFINITE != timeoutms)
+            {
+                unsigned remaining;
+                if (timer.timedout(&remaining))
+                {
+                    StringBuffer timeoutText;
+                    throwStringExceptionV(-1, "Failed to remove %s: %s", logicalname, timeoutException->errorMessage(timeoutText).str());
+                }
+                if (sleepTime>remaining)
```

Comment:
I suppose it doesn't really matter that we could end up waiting timeoutms + SDS_SUB_LOCK_TIMEOUT before failing.
It is only 10 sec more at the most.